src/instance.ts

```diff
@@ -26,9 +28,32 @@ export class ExtensionInstance {
 	constructor(context: ExtensionContext) {
 		this.context = context;
 
-		this.networkComponent = new NetworkComponent(this);
+		const controlInstance = spawn("node", [ 
+			path.join(this.context.extensionPath, "components", "controlComponent.js")
+		])
+
 		this.terminalComponent = new TerminalComponent(this);
  
 		this.onExtensionActivated.connect(() => console.log("Plugin Active!"));
+
+		controlInstance.on('close', (exitStatus) => {
+			exitStatus 
+			? 
+				console.log("controlInstance :: Instance shut down improperly with an error! Abort.")
+			: 
+				console.log("controlInstance :: Shut down API server.")
+		})
+
+		controlInstance.stderr.on('data', (err) => {
+			console.log("controlInstance :: ERROR: ", err)
+		})
+
+		controlInstance.stdout.on('data', (stream) => {
+			console.log("controlInstance :: ", stream)
+		})
+
+		this.onExtensionDeactivated.connect(() => {
```

Comment:
Make this a single line and place it below ln 37 so we're not separating event connections

Reply:
We can only kill the controlInstance once it has instantiated and has event listeners on this. Line 55 is where this is possible.